components/supervisor/pkg/ports/exposed-ports.go

```diff
@@ -174,14 +220,19 @@ func (g *GitpodExposedPorts) doExpose(req *exposePortRequest) {
 
 // Expose exposes a port to the internet. Upon successful execution any Observer will be updated.
 func (g *GitpodExposedPorts) Expose(ctx context.Context, local uint32, public bool) <-chan error {
-	v := "private"
-	if public {
-		v = "public"
+	if !public {
+		if !g.existInLocalExposed(local) {
+			g.localExposedPort = append(g.localExposedPort, local)
+			g.localExposedNotice <- struct{}{}
+		}
+		c := make(chan error)
+		close(c)
+		return c
 	}
 	req := &exposePortRequest{
 		port: &gitpod.WorkspaceInstancePort{
 			Port:       float64(local),
-			Visibility: v,
+			Visibility: "private",
```

Comment:
shouldn't it be public?

```suggestion
			Visibility: "public",
```